### PR TITLE
feat: use of route attributes

### PR DESCRIPTION
**Pull request by asbiin**

### CODE REVIEW

File: app/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers\Api;
```
Comment (djaiss) [inline app/Http/Controllers/Api/UserController.php]: Technically I should this should be under `App\Domains\Settings\ManageUsers\Api` instead

Comment (asbiin) [reply app/Http/Controllers/Api/UserController.php]: make sense

File: app/Domains/Settings/ManageUsers/Api/UserController.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Spatie\RouteAttributes\Attributes\{Get, Middleware, Prefix};
+
+#[Middleware('api')]
+#[Prefix('api')]
+class UserController extends Controller
+{
+    #[Get('user', middleware: 'abilities:read')]
```
Comment (djaiss) [inline app/Domains/Settings/ManageUsers/Api/UserController.php:20]: What is `abilities:read`?

File: pint.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "preset": "laravel",
```
Comment (djaiss) [inline pint.json:2]: It's good to use a dedicated json file for this.

**Review by djaiss — APPROVED** [review]
This is good.

What is this new `app/Http/Controllers/Api/.gitkeep` file in this PR?

Also, how do we manage the nested middlewares that we have? Example: a Post has a Post middleware, a journal middleware and a vault middleware before being able to be read.

File: app/Domains/Settings/ManageUsers/Api/UserController.php (hunk 2 above)
Comment (djaiss) [reply app/Domains/Settings/ManageUsers/Api/UserController.php:20]: Nevermind I saw the answer below.

**Comment by djaiss** [comment]
@asbiin I have some doubts about this. I'm afraid if it will make things more complicated in the long run, as it will be harder to know where is what. Also, it's not standard in a Laravel project so it will be harder for people to onboard.

**Pull request closed without merging**